server/handlers.go

```diff
@@ -254,30 +254,47 @@ func (s *Server) handleConnectorLogin(w http.ResponseWriter, r *http.Request) {
 				s.logger.Errorf("Server template error: %v", err)
 			}
 		case connector.SAMLConnector:
-			action, value, err := conn.POSTData(scopes, authReqID)
+			ssoURL, value, err := conn.POSTData(scopes, authReqID)
```

Comment:
I think it's better to pass http.ResponseWriter and http.Request into provider to reduce provider specific code in server/handlers.go. Eventually we can eliminate this `switch conn := conn.Connector.(type)`

My proposal is to define a common Connector interface including a method:

HandleLogin(w http.ResponseWriter, r *http.Request)

And leave to the Connector implementation for all provider specific logic.

In the mean time, I will do the following change:

Rename PostData to AuthnRequest(w http.ResponseWriter, r *http.Request) and perform the logic there.